flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/KinesisDataFetcher.java

```diff
@@ -272,6 +272,11 @@ private AsyncKinesisRecordEmitter(int queueCapacity) {
 		public void emit(RecordWrapper<T> record, RecordQueue<RecordWrapper<T>> queue) {
 			emitRecordAndUpdateState(record);
 		}
+
+		@Override
+		public void reportError(Throwable throwable) {
+			stopWithError(throwable);
```

Comment:
perhaps still worthwhile to log an error here ?  Call `super.reportError(throwable)` before `stopWithError` ?